patch-package
--- a/node_modules/react-select/dist/react-select.es.js
+++ b/node_modules/react-select/dist/react-select.es.js
@@ -1023,10 +1023,17 @@ var Select$1 = function (_React$Component) {
 				this.props.onFocus(event);
 			}
 
-			this.setState({
+			var onFocusState = {
 				isFocused: true,
 				isOpen: !!toOpen
-			});
+			};
+			if (this.props.keepValueOnFocus && !this.props.multi && this.props.value && !this._focusAfterClear) {
+				var valueStr = typeof this.props.value === 'string' ? this.props.value : this.props.value.label;
+				if (valueStr) {
+					onFocusState.inputValue = this.handleInputValueChange(valueStr);
+				}
+			}
+			this.setState(onFocusState);
 
 			this._focusAfterClear = false;
 			this._openAfterFocus = false;
@@ -1203,6 +1210,18 @@ var Select$1 = function (_React$Component) {
 	}, {
 		key: 'handleValueClick',
 		value: function handleValueClick(option, event) {
+			var _this2 = this;
+
+			if (this.props.editValueIfClicked) {
+				this.removeValue(option);
+				setTimeout(function () {
+					_this2.setState({
+						inputValue: _this2.handleInputValueChange(option.value),
+						isOpen: false
+					}, _this2.focus);
+				}, 1);
+			}
+
 			if (!this.props.onValueClick) return;
 			this.props.onValueClick(option, event);
 		}
@@ -1256,7 +1275,7 @@ var Select$1 = function (_React$Component) {
 	}, {
 		key: 'setValue',
 		value: function setValue(value) {
-			var _this2 = this;
+			var _this3 = this;
 
 			if (this.props.autoBlur) {
 				this.blurInput();
@@ -1267,7 +1286,7 @@ var Select$1 = function (_React$Component) {
 			}
 			if (this.props.simpleValue && value) {
 				value = this.props.multi ? value.map(function (i) {
-					return i[_this2.props.valueKey];
+					return i[_this3.props.valueKey];
 				}).join(this.props.delimiter) : value[this.props.valueKey];
 			}
 			if (this.props.onChange) {
@@ -1277,7 +1296,7 @@ var Select$1 = function (_React$Component) {
 	}, {
 		key: 'selectValue',
 		value: function selectValue(value) {
-			var _this3 = this;
+			var _this4 = this;
 
 			// NOTE: we actually add/set the value in a callback to make sure the
 			// input value is empty to avoid styling issues in Chrome
@@ -1291,13 +1310,13 @@ var Select$1 = function (_React$Component) {
 					inputValue: this.handleInputValueChange(updatedValue),
 					isOpen: !this.props.closeOnSelect
 				}, function () {
-					var valueArray = _this3.getValueArray(_this3.props.value);
+					var valueArray = _this4.getValueArray(_this4.props.value);
 					if (valueArray.some(function (i) {
-						return i[_this3.props.valueKey] === value[_this3.props.valueKey];
+						return i[_this4.props.valueKey] === value[_this4.props.valueKey];
 					})) {
-						_this3.removeValue(value);
+						_this4.removeValue(value);
 					} else {
-						_this3.addValue(value);
+						_this4.addValue(value);
 					}
 				});
 			} else {
@@ -1306,7 +1325,13 @@ var Select$1 = function (_React$Component) {
 					isOpen: !this.props.closeOnSelect,
 					isPseudoFocused: this.state.isFocused
 				}, function () {
-					_this3.setValue(value);
+					_this4.setValue(value);
+					if (_this4.props.keepValueOnFocus) {
+						_this4.blurInput();
+						setTimeout(function () {
+							return _this4.input.focus();
+						});
+					}
 				});
 			}
 		}
@@ -1338,11 +1363,11 @@ var Select$1 = function (_React$Component) {
 	}, {
 		key: 'removeValue',
 		value: function removeValue(value) {
-			var _this4 = this;
+			var _this5 = this;
 
 			var valueArray = this.getValueArray(this.props.value);
 			this.setValue(valueArray.filter(function (i) {
-				return i[_this4.props.valueKey] !== value[_this4.props.valueKey];
+				return i[_this5.props.valueKey] !== value[_this5.props.valueKey];
 			}));
 			this.focus();
 		}
@@ -1503,7 +1528,7 @@ var Select$1 = function (_React$Component) {
 	}, {
 		key: 'renderValue',
 		value: function renderValue(valueArray, isOpen) {
-			var _this5 = this;
+			var _this6 = this;
 
 			var renderLabel = this.props.valueRenderer || this.getOptionLabel;
 			var ValueComponent = this.props.valueComponent;
@@ -1521,13 +1546,13 @@ var Select$1 = function (_React$Component) {
 					return React.createElement(
 						ValueComponent,
 						{
-							disabled: _this5.props.disabled || value.clearableValue === false,
-							id: _this5._instancePrefix + '-value-' + i,
-							instancePrefix: _this5._instancePrefix,
-							key: 'value-' + i + '-' + value[_this5.props.valueKey],
+							disabled: _this6.props.disabled || value.clearableValue === false,
+							id: _this6._instancePrefix + '-value-' + i,
+							instancePrefix: _this6._instancePrefix,
+							key: 'value-' + i + '-' + value[_this6.props.valueKey],
 							onClick: onClick,
-							onRemove: _this5.removeValue,
-							placeholder: _this5.props.placeholder,
+							onRemove: _this6.removeValue,
+							placeholder: _this6.props.placeholder,
 							value: value
 						},
 						renderLabel(value, i),
@@ -1558,7 +1583,7 @@ var Select$1 = function (_React$Component) {
 		key: 'renderInput',
 		value: function renderInput(valueArray, focusedOptionIndex) {
 			var _classNames,
-			    _this6 = this;
+			    _this7 = this;
 
 			var className = classNames('Select-input', this.props.inputProps.className);
 			var isOpen = this.state.isOpen;
@@ -1584,7 +1609,7 @@ var Select$1 = function (_React$Component) {
 				onChange: this.handleInputChange,
 				onFocus: this.handleInputFocus,
 				ref: function ref(_ref) {
-					return _this6.input = _ref;
+					return _this7.input = _ref;
 				},
 				role: 'combobox',
 				required: this.state.required,
@@ -1612,7 +1637,7 @@ var Select$1 = function (_React$Component) {
 					onBlur: this.handleInputBlur,
 					onFocus: this.handleInputFocus,
 					ref: function ref(_ref2) {
-						return _this6.input = _ref2;
+						return _this7.input = _ref2;
 					},
 					role: 'combobox',
 					style: { border: 0, width: 1, display: 'inline-block' },
@@ -1738,18 +1763,18 @@ var Select$1 = function (_React$Component) {
 	}, {
 		key: 'renderHiddenField',
 		value: function renderHiddenField(valueArray) {
-			var _this7 = this;
+			var _this8 = this;
 
 			if (!this.props.name) return;
 			if (this.props.joinValues) {
 				var value = valueArray.map(function (i) {
-					return stringifyValue(i[_this7.props.valueKey]);
+					return stringifyValue(i[_this8.props.valueKey]);
 				}).join(this.props.delimiter);
 				return React.createElement('input', {
 					disabled: this.props.disabled,
 					name: this.props.name,
 					ref: function ref(_ref3) {
-						return _this7.value = _ref3;
+						return _this8.value = _ref3;
 					},
 					type: 'hidden',
 					value: value
@@ -1757,12 +1782,12 @@ var Select$1 = function (_React$Component) {
 			}
 			return valueArray.map(function (item, index) {
 				return React.createElement('input', {
-					disabled: _this7.props.disabled,
+					disabled: _this8.props.disabled,
 					key: 'hidden.' + index,
-					name: _this7.props.name,
+					name: _this8.props.name,
 					ref: 'value' + index,
 					type: 'hidden',
-					value: stringifyValue(item[_this7.props.valueKey])
+					value: stringifyValue(item[_this8.props.valueKey])
 				});
 			});
 		}
@@ -1796,7 +1821,7 @@ var Select$1 = function (_React$Component) {
 	}, {
 		key: 'renderOuter',
 		value: function renderOuter(options, valueArray, focusedOption) {
-			var _this8 = this;
+			var _this9 = this;
 
 			var menu = this.renderMenu(options, valueArray, focusedOption);
 			if (!menu) {
@@ -1806,7 +1831,7 @@ var Select$1 = function (_React$Component) {
 			return React.createElement(
 				'div',
 				{ ref: function ref(_ref5) {
-						return _this8.menuContainer = _ref5;
+						return _this9.menuContainer = _ref5;
 					}, className: 'Select-menu-outer', style: this.props.menuContainerStyle },
 				React.createElement(
 					'div',
@@ -1816,7 +1841,7 @@ var Select$1 = function (_React$Component) {
 						onMouseDown: this.handleMouseDownOnMenu,
 						onScroll: this.handleMenuScroll,
 						ref: function ref(_ref4) {
-							return _this8.menu = _ref4;
+							return _this9.menu = _ref4;
 						},
 						role: 'listbox',
 						style: this.props.menuStyle,
@@ -1829,7 +1854,7 @@ var Select$1 = function (_React$Component) {
 	}, {
 		key: 'render',
 		value: function render() {
-			var _this9 = this;
+			var _this10 = this;
 
 			var valueArray = this.getValueArray(this.props.value);
 			var options = this._visibleOptions = this.filterOptions(this.props.multi && this.props.removeSelected ? valueArray : null);
@@ -1869,7 +1894,7 @@ var Select$1 = function (_React$Component) {
 			return React.createElement(
 				'div',
 				{ ref: function ref(_ref7) {
-						return _this9.wrapper = _ref7;
+						return _this10.wrapper = _ref7;
 					},
 					className: className,
 					style: this.props.wrapperStyle },
@@ -1877,7 +1902,7 @@ var Select$1 = function (_React$Component) {
 				React.createElement(
 					'div',
 					{ ref: function ref(_ref6) {
-							return _this9.control = _ref6;
+							return _this10.control = _ref6;
 						},
 						className: 'Select-control',
 						onKeyDown: this.handleKeyDown,
